receivers/sns/config.go

```
@@ -0,0 +1,104 @@
+		Sigv4       SigV4Config       `yaml:"sigv4" json:"sigv4"`
+		TopicARN    string            `yaml:"topic_arn,omitempty" json:"topic_arn,omitempty"`
+		PhoneNumber string            `yaml:"phone_number,omitempty" json:"phone_number,omitempty"`
+		TargetARN   string            `yaml:"target_arn,omitempty" json:"target_arn,omitempty"`
+		Subject     string            `yaml:"subject,omitempty" json:"subject,omitempty"`
+		Message     string            `yaml:"message,omitempty" json:"message,omitempty"`
+		Attributes  map[string]string `yaml:"attributes,omitempty" json:"attributes,omitempty"`
+	}
+	var settings snsSettingsRaw
+	err := json.Unmarshal(jsonData, &settings)
+	if err != nil {
+		return Config{}, fmt.Errorf("failed to unmarshal settings: %w", err)
+	}
+
+	if settings.TopicARN != "" {
+		_, err = arn.Parse(settings.TopicARN)
+		if err != nil {
+			return Config{}, errors.New("invalid topic ARN provided")
+		}
+	}
+
+	if settings.TargetARN != "" {
+		_, err = arn.Parse(settings.TopicARN)
+		if err != nil {
+			return Config{}, errors.New("invalid target ARN provided")
+		}
+	}
+
+	if settings.TopicARN == "" && settings.TargetARN == "" && settings.PhoneNumber == "" {
+		return Config{}, errors.New("must specify topicArn, targetArn, or phone number")
+	}
+	if settings.Subject == "" {
+		settings.Subject = templates.DefaultMessageTitleEmbed
+	}
+	if settings.Message == "" {
+		settings.Message = templates.DefaultMessageEmbed
+	}
+	if settings.APIUrl == "" {
+		settings.APIUrl = fmt.Sprintf("https://sns.%s.amazonaws.com", settings.Sigv4.Region)
+	}
+
+	at := awsds.AuthTypeDefault
+	if settings.Sigv4.Profile != "" {
+		at = awsds.AuthTypeSharedCreds
+	} else if settings.Sigv4.AccessKey != "" && settings.Sigv4.SecretKey != "" {
+		settings.Sigv4.AccessKey = decryptFn("accessKey", settings.Sigv4.AccessKey)
+		settings.Sigv4.SecretKey = decryptFn("secretKey", settings.Sigv4.SecretKey)
+		at = awsds.AuthTypeKeys
+	}
```

Can we add a validation here to check for situations when either of keys is not specified.  Currently, it silently falls back to default authentication type

---

Yes, will do.